ndscan/experiment/default_analysis.py

```diff
@@ -36,6 +43,49 @@ def __init__(self, kind: str, **kwargs):
         self.spec = {"kind": kind, **kwargs}
 
 
+class AnalysisType(Enum):
+    NAMED_FIT = 'named_fit'
+
+
+@dataclasses.dataclass
+class Analysis:
+    kind: AnalysisType = dataclasses.field(init=False)
+
+    def __post_init__(self):
+        if type(self) == NamedFit:
+            self.kind = AnalysisType.NAMED_FIT
+
+
+@dataclasses.dataclass
+class NamedFit(Analysis):
+    """ Named fit function used in online analyses.
+
+    Attributes:
+        fit_class_name: name of the fit class to use
+        fit_module: module that fit_class resides in (must be in the current python path)
+        data: maps fit data axis names (``"x"``, ``"y"``) to parameter handles or result
+            channels that supply the respective data.
+        param_bounds: dictionary of tuples containing the lower and upper bounds for
+            each parameter. If not specified, the defaults from the fit class are used.
+        fixed_params: dictionary specifying constant values for any non-floated
+            parameters. If not specified, the defaults from the fit class are used.
+        initial_values: dictionary specifying initial parameter values to use in
+            the fit. These override the values found by the fit's parameter estimator.
+        scale_factors: dictionary specifying scale factors for parameters. The
```

Comment:
I did briefly consider having a mechanism that allows the user to get default scale factors from the axis units (e.g. if I'm scanning a time in `us` then `1e6` is probably a good scale factor) but this felt more effort than it's worth! We could also do something like the default scale factor estimation in `oitg.fitting` but AFAICT this can do nasty things if the initial parameter guess is close to (but not equal to) zero -- although I might be wrong, I haven't fully understood that code!